dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -251,73 +253,82 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                     &cookie));
             }
 
-        if (WI_IsFlagSet(registrationOptions, PushNotificationRegistrationActivators::ComActivator))
-        {
+            if (builder)
             {
-                auto lock = s_lock.lock_exclusive();
-                THROW_HR_IF_MSG(E_INVALIDARG, s_cookie > 0, "ComActivator already registered.");
+                registeredTaskFromBuilder = builder.Register();
             }
 
-            GetWaitHandleForArgs().create();
-
-            THROW_IF_FAILED(::CoRegisterClassObject(
-                taskClsid,
-                winrt::make<PushNotificationBackgroundTaskFactory>().get(),
-                CLSCTX_LOCAL_SERVER,
-                REGCLS_MULTIPLEUSE,
-                &cookie));
-        }
-
-            PushNotificationRegistrationToken token = { cookie, registeredTaskFromBuilder };
             scopeExitToCleanRegistrations.release();
 
-            PushNotificationTelemetry::ActivatorRegisteredByApi(S_OK, details.Options());
+            auto lock = s_activatorInfoLock.lock_exclusive();
+            s_cookie.reset(cookie);
```

Comment:
This is wrong. The usage of this wil wrapper should be something like:

`CoRegisterClassObject(..., &s_cookie);`

Then you just call `reset()` whenever you need to revoke the class object.

Having said that, probably it's not `s_cookie` but `s_comActivatorRegistration` or similar;